run_integration_tests.go

```diff
@@ -0,0 +1,189 @@
+	// run fuse tests on darwin
+	if runtime.GOOS != "darwin" {
+		msg("skip fuse integration tests on %v\n", runtime.GOOS)
+		os.Setenv("RESTIC_TEST_FUSE", "0")
+	}
+
+	// compile for all target architectures with tags
+	for _, tags := range []string{"release", "debug"} {
+		run("gox", "-verbose",
+			"-os", strings.Join(env.goxOS, " "),
+			"-arch", strings.Join(env.goxArch, " "),
+			"-tags", tags,
+			"./cmd/restic")
+	}
+
+	// run the build script
+	run("go", "run", "build.go")
+
+	// gather coverage information
+	run("go", "run", "run_tests.go", "all.cov")
+
+	runGofmt()
+}
+
+type AppveyorEnvironment struct{}
+
+func (env *AppveyorEnvironment) Prepare() {
+	msg("preparing environment for Appveyor CI\n")
+
+	// install tar, gzip, bzip2
+}
+
+func (env *AppveyorEnvironment) RunTests() {
+	// run the build script and the tests
+	run("go", "run", "build.go", "-v", "-T")
+}
+
+// findGoFiles returns a list of go source code file names below dir.
+func findGoFiles(dir string) (list []string, err error) {
+	err = filepath.Walk(dir, func(name string, fi os.FileInfo, err error) error {
+		if filepath.Base(name) == "Godeps" {
+			return filepath.SkipDir
+		}
+
+		if filepath.Ext(name) == ".go" {
+			relpath, err := filepath.Rel(dir, name)
+			if err != nil {
+				return err
+			}
+
+			list = append(list, relpath)
+		}
+
+		return err
+	})
+
+	return list, err
+}
+
+func msg(format string, args ...interface{}) {
+	fmt.Printf("CI: "+format, args...)
+}
+
+func runGofmt() {
+	dir, err := os.Getwd()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Getwd(): %v\n", err)
+		os.Exit(5)
+	}
+
+	files, err := findGoFiles(dir)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error finding Go files: %v\n", err)
+		os.Exit(4)
+	}
+
+	msg("runGofmt() with %d files\n", len(files))
+	args := append([]string{"-l"}, files...)
+	cmd := exec.Command("gofmt", args...)
+	cmd.Stderr = os.Stderr
+
+	buf, err := cmd.Output()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error running gofmt: %v", err)
+		fmt.Fprintf(os.Stderr, "output:\n%s\n", buf)
+		os.Exit(3)
+	}
+
+	if len(buf) > 0 {
+		fmt.Fprintf(os.Stderr, "not formatted with `gofmt`:\n")
+		fmt.Fprintln(os.Stderr, string(buf))
+		os.Exit(6)
+	}
+}
+
+func run(command string, args ...string) {
+	msg("run %v %v\n", command, strings.Join(args, " "))
+	cmd := exec.Command(command, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error running %v %v: %v",
+			command, strings.Join(args, " "), err)
+		os.Exit(3)
+	}
+}
+
+func isTravis() bool {
+	return os.Getenv("TRAVIS_BUILD_DIR") != ""
+}
+
+func isAppveyor() bool {
+	return runtime.GOOS == "windows"
+}
+
+func main() {
+	var env CIEnvironment
+
+	switch {
+	case isTravis():
+		env = &TravisEnvironment{}
+	case isAppveyor():
+		env = &AppveyorEnvironment{}
+	default:
+		fmt.Fprintln(os.Stderr, "unknown CI environment")
+		os.Exit(1)
+	}
+
+	for _, f := range []func(){env.Prepare, env.RunTests} {
```

Comment:
how about making those two things (prepare and run) two separate commands (using a command line argument switch or something)? then you could do those two steps separately in the CI environment and at least make use of the separate output grouping etc., and it would also make it easier to run only one but not the other (maybe I want to run the tests locally, but I don't need to prepare locally).